sw/device/sca/ecc384_serial.c

```diff
@@ -0,0 +1,345 @@
+                            const uint8_t *private_key_d, uint8_t *signature_r,
+                            uint8_t *signature_s, const uint8_t *k) {
+  CHECK(otbn_ctx != NULL);
+
+  // Set pointers to input arguments.
+  LOG_INFO("Setup data pointers");
+  setup_data_pointers(otbn_ctx);
+
+  // Write input arguments.
+  uint32_t mode = 1;  // mode 1 => sign
+  LOG_INFO("Copy data");
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, sizeof(mode), &mode, kOtbnVarMode) ==
+        kOtbnOk);
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, /*len_bytes=*/48, msg, kOtbnVarMsg) ==
+        kOtbnOk);
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, /*len_bytes=*/48, private_key_d,
+                               kOtbnVarD) == kOtbnOk);
+
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, /*len_bytes=*/48, k, kOtbnVarK) ==
+        kOtbnOk);
+
+  // Call OTBN to perform operation, and wait for it to complete.
+  LOG_INFO("Execute");
+  CHECK(otbn_execute(otbn_ctx) == kOtbnOk);
+  LOG_INFO("Wait for done");
+  CHECK(otbn_busy_wait_for_done(otbn_ctx) == kOtbnOk);
+
+  // Read back results.
+  LOG_INFO("Get results");
+  CHECK(otbn_copy_data_from_otbn(otbn_ctx, /*len_bytes=*/48, kOtbnVarR,
+                                 signature_r) == kOtbnOk);
+  CHECK(otbn_copy_data_from_otbn(otbn_ctx, /*len_bytes=*/48, kOtbnVarS,
+                                 signature_s) == kOtbnOk);
+  LOG_INFO("r[0]: 0x%02x", signature_r[0]);
+  LOG_INFO("s[0]: 0x%02x", signature_s[0]);
+}
+
+/**
+ * Simple serial 'p' (sign) command handler.
+ *
+ * Takes the scalar value from the simple serial UART
+ * and triggers OTBN_P384_sign operation.
+ *
+ * To overwrite the message, use the simpleserial command 'n'
+ * To overwrite the private key value, use the simpleserial command 'd'
+ *
+ * @param ecc384_secret_k the ephemeral key/scalar provided via simpleserial
+ * UART.
+ * @param secret_k_len Length of the ephemeral key.
+ */
+static void ecc384_simpleserial_ecdsa(const uint8_t *ecc384_secret_k,
+                                      size_t secret_k_len) {
+  otbn_t otbn_ctx;
+
+  /* ecc384_secret_k = ephemeral key k, coming from UART */
+  if (secret_k_len != 48) {
+    LOG_INFO("Invalid data length %hu", (uint8_t)secret_k_len);
+    return;
+  }
+  LOG_INFO("SSECDSA starting...");
+  CHECK(otbn_init(&otbn_ctx, mmio_region_from_addr(
+                                 TOP_EARLGREY_OTBN_BASE_ADDR)) == kOtbnOk);
+
+  CHECK(otbn_zero_data_memory(&otbn_ctx) == kOtbnOk);
+  CHECK(otbn_load_app(&otbn_ctx, kOtbnAppP384Ecdsa) == kOtbnOk);
+
+  uint8_t ecc384_signature_r[48] = {0};
+  uint8_t ecc384_signature_s[48] = {0};
+
+  /* Sign */
+  LOG_INFO("Signing");
+  sca_set_trigger_high();
+  p384_ecdsa_sign(&otbn_ctx, ecc384_msg, ecc384_private_key_d,
+                  ecc384_signature_r, ecc384_signature_s, ecc384_secret_k);
+  sca_set_trigger_low();
+
+  /**
+   * Send the signature_r and signature_s back to host
+   *
+   * TODO: Remove them if they are not necessary for the side-channel analysis.
```

Comment:
Is this TODO intended to remain?

Reply:
For now, this is fine I think. I've added an item to #13905 to track the implementation of a golden model on the host side.